From 62fa823a32cbc9fe7e3fe19a90112a5676ac7b9e Mon Sep 17 00:00:00 2001
From: Nan Wu <wnan@google.com>
Date: Mon, 15 Sep 2025 14:13:46 +0000
Subject: [PATCH] RESTRICT AUTOMERGE: Backport Cut max duration for default
 transition to 1500

3 seconds is way too long, especially for tap jacking exploitation during animation. Cut it down to 1500 for a more tolerable experience.

DISABLE_TOPIC_PROTECTOR

Bug: 266433089
Test: manual
Flag: EXEMPT BUGFIX
Change-Id: I5afd5c942b8f96347398a9b0aaafda28dc83d789
---
 .../android/wm/shell/transition/DefaultTransitionHandler.java   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java b/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java
index e52fd00e7df7..c111d040d491 100644
--- a/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java
+++ b/libs/WindowManager/Shell/src/com/android/wm/shell/transition/DefaultTransitionHandler.java
@@ -118,7 +118,7 @@ import java.util.function.Consumer;
 
 /** The default handler that handles anything not already handled. */
 public class DefaultTransitionHandler implements Transitions.TransitionHandler {
-    private static final int MAX_ANIMATION_DURATION = 3000;
+    private static final int MAX_ANIMATION_DURATION = 1500;
 
     private final TransactionPool mTransactionPool;
     private final DisplayController mDisplayController;
-- 
2.52.0.rc1.455.g30608eb744-goog

